src/pybind/mgr/dashboard/frontend/src/app/shared/api/prometheus.service.ts

```diff
@@ -48,8 +49,28 @@ export class PrometheusService {
     return this.http.get<AlertmanagerSilence[]>(`${this.baseURL}/silences`, { params });
   }
 
-  getRules(params = {}): Observable<PrometheusRule[]> {
-    return this.http.get<PrometheusRule[]>(`${this.baseURL}/rules`, { params });
+  getRules(
+    type: 'all' | 'alerting' | 'rewrites' = 'all'
+  ): Observable<{ groups: PrometheusRuleGroup[] }> {
+    let rules = this.http.get<{ groups: PrometheusRuleGroup[] }>(`${this.baseURL}/rules`);
+    const filterByType = (_type: 'alerting' | 'rewrites') => {
```

Comment:
Wouldn't it be more elegant to use [RxJS filter](https://www.learnrxjs.io/operators/filtering/filter.html) here?

Reply:
Well, I use [RxJS map](https://www.learnrxjs.io/operators/transformation/map.html) here but require to iterate the `_rules` variable inside that map. I don't know how to do that with `filter` from RxJS as I don't think I can get to the same result using `filter` after map has been used, because I require `filter` to be used inside that map. But I'm curious to learn how that's possible. `filterByType` is just outsourced for the sake of clarity. Can you please elaborate?

Reply:
A simplified implementation can be found here: https://github.com/ceph/ceph/pull/32130